src/librustc/ty/maps.rs

```diff
@@ -359,8 +365,8 @@ macro_rules! define_maps {
                 }
 
                 // FIXME(eddyb) Get more valid Span's on queries.
-                if span == DUMMY_SP {
-                    span = key.default_span(tcx);
+                if span == DUMMY_SP && stringify!($name) != "def_span" {
```

Comment:
I guess this is not too bad of a workaround - could you leave a comment above as to why this is done?